### PR TITLE
Add train summary script

### PR DESCRIPTION
**Pull request by rfk**
I've been using an out-of-tree script to generate my train summary emails, but there's no reason why it shouldn't actually live in the repo here with our other scripting.  This PR introduces a lot of new capabilities to the github automation lib, and its fair share of hackiness, but hopefully it's not *too* hard to follow...@mozilla/fxa-devs r?

### CODE REVIEW

File: scripts/audit_collaborators.js [hunk 1]
```diff
@@ -26,10 +26,10 @@ module.exports = {
   },
 
   auditCollaborators: function auditCollaborators(gh, collaborators, repo) {
-    return gh.repos.getCollaborators({ repo: repo }).then(function(users) {
+    return gh.repos.getCollaborators({ repo: repo.name }).then(function(users) {
```
Comment (rfk) [inline scripts/audit_collaborators.js]: Repos are now identified by `{owner, name}` pairs rather than just string name.  These calls should be updated to use the same convention, but that was out-of-scope for importing the new script.

File: scripts/gh.js [hunk 2]
```diff
@@ -1,93 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-//  High-level github automation API.
```
Comment (rfk) [inline scripts/gh.js]: I renamed this to "ghlib" to clear up some ugliness during import/instantiation, where he'd have things like "gh = new GH.GH()".

File: scripts/train_summary.js [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  'fxa-profile-server': {},
+  'fxa-basket-proxy': {},
+}
+
+console.log("Updating repos...")
+ghlib.ensureLocalRepos()
+.then(function() {
+
+  return P.each(ghlib.REPOS, function(repo) {
+    // For now, we're only looking at reposo we actually tag-and-release from.
```
Comment (philbooth) [inline scripts/train_summary.js]: Typo `reposo`.

**Comment by philbooth** [comment]
Looks fine to me, r+.
